temp: test XBlock==6, which adds type annotations

kdmccormick opened the pull request:
Testing https://github.com/openedx/XBlock/pull/761

kdmccormick committed:
temp: test XBlock==6, which adds type annotations

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35236-7daceb.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35236-7daceb.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7517362556-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7517362556-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7517362556-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35236-7daceb.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35236-7daceb.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7558719427-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7558719427-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7558719427-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35236-7daceb.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35236-7daceb.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7559048306-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7559048306-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7559048306-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35236-7daceb.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35236-7daceb.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7622724234-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7622724234-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7622724234-tutor-requirements.txt)

kdmccormick committed:
refactor: make XModuleMixin extend XBlockMixin, not XBlock

In order to get typechecking to pass in XBlock, we needed to make
ScorableXBlockMixin a subclass of XBlockMixin. This change, while
sensible, creates an ambiguous method resolution order in the
ProblemBlock, stemming from the fact that XModuleMixin inherits directly
from XBlock.

The fix is to simply make XModuleMixin inherit from XBlockMixin
instead-- this makes sense because XModuleMixin is a helper mixin, not
an XBlock itself. We then must add XBlock as an explicit subclass to all
blocks which inherit from XBlock. This should have no functional impact,
but *should* help us add type hints to the xmodule/ folder in the
future.

Using ProblemBlock as an example... before (XBlock < 6.0.0) we have:

  ProblemBlock <- ScorableXBlockMixin
               <- XModuleMixin <- XBlock <- Blocklike

after (XBlock >= 6.0.0) we have:

  ProblemBlock <- ScorableXBlockMixin <- XBlockMixin <- Blocklike
               <- XModuleMixin        <- XBlockMixin <- Blocklike
               <- XBlock                             <- Blocklike

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35236-7daceb.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35236-7daceb.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7624475210-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7624475210-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7624475210-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35236-7daceb.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35236-7daceb.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7625168474-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7625168474-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7625168474-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35236-7daceb.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35236-7daceb.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7625370641-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7625370641-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7625370641-tutor-requirements.txt)

open-craft-grove commented:
Sandbox deployment successful 🚀
🎓 [LMS](https://pr-35236-7daceb.sandboxes.opencraft.hosting)
📝 [Studio](https://studio.pr-35236-7daceb.sandboxes.opencraft.hosting)
ℹ️ [Grove Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7625580948-tutor-config.yml), [Tutor Config](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7625580948-grove-config.yml), [Tutor Requirements](https://axim-sandbox-build-logs-4626178240.nyc3.digitaloceanspaces.com/pr-35236-7daceb-51564656-7625580948-tutor-requirements.txt)
